portability-core/src/main/java/org/dataportabilityproject/shared/ServiceProvider.java

```diff
@@ -46,37 +46,19 @@
      * have permissions based on the serviceMode provided.
      */
     default OfflineAuthDataGenerator getOfflineAuthDataGenerator(PortableDataType dataType, ServiceMode serviceMode) {
-      // default implementation to ignore serviceMode
-      return getOfflineAuthDataGenerator(dataType);
+      logger.warn("WARNING: getOnlineAuthDataGenerator not implemented for type: {}, serviceMode: {}, service: {}",
```

Comment:
One last thing, you can remove the WARNING: from the log statement as the warn() method will prepend that to the output. Here and below.

Reply:
Good catch! Done.